ai21/clients/bedrock/_stream_decoder.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import json
+from functools import lru_cache
+from typing import Iterator, AsyncIterator
+
+import httpx
+from botocore.eventstream import EventStreamMessage
+from botocore.model import Shape
+
+from ai21.errors import StreamingDecodeError
+
+
+@lru_cache(maxsize=None)
+def get_response_stream_shape() -> Shape:
+    from botocore.model import ServiceModel
+    from botocore.loaders import Loader
+
+    loader = Loader()
+    bedrock_service_dict = loader.load_service_model("bedrock-runtime", "service-2")
+    bedrock_service_model = ServiceModel(bedrock_service_dict)
+    return bedrock_service_model.shape_for("ResponseStream")
+
+
+class AWSEventStreamDecoder:
+    def __init__(self) -> None:
+        from botocore.parsers import EventStreamJSONParser
```

Comment:
Why is this import in the ctor? I see you're importing boto stuff outside of the class as well

Reply:
Done